rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistStorageProviderImplTest.java

```diff
@@ -0,0 +1,216 @@
+
+        whitelistStorageProvider = new WhitelistStorageProviderImpl(inMemoryStorage);
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        assertEquals(1, actualLockWhitelist.getAll().size());
+
+        //Making sure the saved value is correct and related to OneOffWhiteListEntry
+        Map<Address, OneOffWhiteListEntry> oneOffWhiteListEntryMap = getAddressFromOneOffWhiteListEntryInMemoryStorage();
+        assertTrue(oneOffWhiteListEntryMap.containsKey(firstBtcAddress));
+
+        //Making sure there is no value saved in storage related to UnlimitedWhiteListEntry
+        Map<Address, UnlimitedWhiteListEntry> unlimitedWhiteListEntryMap = getAddressFromUnlimitedWhiteListEntryInMemoryStorage();
+        assertEquals(0, unlimitedWhiteListEntryMap.size());
+    }
+
+    @Test
+    void save_whenIsActiveRSKIP87_ShouldSavedUnlimitedLockWhitelist() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(true);
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        //make sure the lockWhitelist is empty
+        assertEquals(0, lockWhitelist.getAll().size());
+
+        LockWhitelistEntry oneOffWhiteListEntry =  createOneOffWhiteListEntry(firstBtcAddress);
+        LockWhitelistEntry unlimitedWhiteListEntry = createUnlimitedWhiteListEntry(secondBtcAddress);
+        lockWhitelist.put(firstBtcAddress, oneOffWhiteListEntry);
+        lockWhitelist.put(secondBtcAddress, unlimitedWhiteListEntry);
+
+        whitelistStorageProvider.save(activationConfig);
+
+        whitelistStorageProvider = new WhitelistStorageProviderImpl(inMemoryStorage);
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        assertEquals(2, actualLockWhitelist.getAll().size());
+
+        //Making sure the saved value is correct
+        Map<Address, OneOffWhiteListEntry> oneOffWhiteListEntryMap = getAddressFromOneOffWhiteListEntryInMemoryStorage();
+        assertTrue(oneOffWhiteListEntryMap.containsKey(firstBtcAddress));
+
+        Map<Address, UnlimitedWhiteListEntry> lockWhitelistEntryMap = getAddressFromUnlimitedWhiteListEntryInMemoryStorage();
+        assertTrue(lockWhitelistEntryMap.containsKey(secondBtcAddress));
+    }
+
+    @Test
+    void getLockWhitelist_whenNoEntriesInStorage_shouldReturnZeroEntries() {
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        assertEquals(0, actualLockWhitelist.getAll().size());
+    }
+
+    @Test
+    void getLockWhitelist_whenSavedValueDirectlyInStorageAndLockWhitelistIsNotNull_shouldReturnZeroEntries() {
+        //The first time LockWhitelist is null and it queries the storage
+        LockWhitelist firstTimeQuery = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
```

Comment:
i think `firstTimeQuery` is not a super-descriptive name, wdyt about something like `lockWhitelistFromEmptyStorage` ?

Reply:
or better, something like

```
LockWhitelist lockWhitelist;
// The first time LockWhitelist is null and it queries the storage
lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
```

Reply:
Do you mean to create the `lockWhitelist` as a kind of temporary variable? Could it make sense to call it `tempLockWhitelist`? WDYT?

Reply:
i dont see any benefits of calling it `tempLockWhitelist`, aren't most of variables temporary? jaja
i'd simply call it `lockWhitelist`.

btw, to avoid doing [this](https://github.com/rsksmart/rskj/pull/2498/files/3ee384652a2b3b342a702c7f480acc68f210fc8f#diff-4c11f95737338a52ebfaae1f3c570691c244e8eb14eedb9888743489488fd06cR144) you can do [what jere did](https://github.com/rsksmart/rskj/pull/2463/files#diff-f42b277f39730e846e15f899b88e1e7e718280d3ea3d4c4d5b035a5d66385923R634) (getting the value directly from repository)

Reply:
The benefit is related to maintainability. It helps to understand at a glance what the variable's function is, and that it'll be updated several times.

Reply:
For the second suggestion, we need to recreate the provider, indeed, it is part of the challenge in this test. You can go deeper into it on this thread:
https://github.com/rsksmart/rskj/pull/2498#discussion_r1649755505

Reply:
but you are doing it so its query the storage, right? 
If you query the storage directly you avoid the recreation.

Reply:
Done! Created the `tempLockWhitelist` variable. Nevertheless, we can discuss the variable name further to get an agreement about this topic.

Reply:
But, we want the recreation to get the data from storage using `whitelistStorageProvider.getLockWhitelist`.